pytorch_lightning/utilities/grads.py

```diff
@@ -35,18 +35,13 @@ def grad_norm(module: Module, norm_type: Union[float, int, str]) -> Dict[str, fl
             as a single vector.
     """
     norm_type = float(norm_type)
-
-    norms, all_norms = {}, []
-    for name, p in module.named_parameters():
-        if p.grad is None:
-            continue
-
-        param_norm = float(p.grad.data.norm(norm_type))
-        norms[f"grad_{norm_type}_norm_{name}"] = round(param_norm, 4)
-
-        all_norms.append(param_norm)
-
-    total_norm = float(torch.tensor(all_norms).norm(norm_type))
-    norms[f"grad_{norm_type}_norm_total"] = round(total_norm, 4)
```

Comment:
Improvement: `torch.tensor([]).norm()` is equal to `tensor(0.)` so when there are no gradients, this code stills adds the key for total norm (0).